implementations/rust/ockam/ockam_node/src/router/record.rs

```diff
@@ -162,21 +162,27 @@ impl InternalMap {
     }
 
     /// Retrieve the next cluster in reverse-initialisation order
+    /// Return None if there is no next cluster or if the cluster
+    /// contained no more active addresses
     pub(super) fn next_cluster(&mut self) -> Option<Vec<&mut AddressRecord>> {
         let name = self.cluster_order.pop()?;
         let addrs = self.clusters.remove(&name)?;
-        Some(
-            self.address_records_map
-                .iter_mut()
-                .filter_map(|(primary, rec)| {
-                    if addrs.contains(primary) {
-                        Some(rec)
-                    } else {
-                        None
-                    }
-                })
-                .collect(),
-        )
+        let active_addresses: Vec<&mut AddressRecord> = self
+            .address_records_map
+            .iter_mut()
+            .filter_map(|(primary, rec)| {
+                if addrs.contains(primary) {
+                    Some(rec)
+                } else {
+                    None
+                }
+            })
+            .collect();
+        if active_addresses.is_empty() {
+            None
+        } else {
+            Some(active_addresses)
+        }
```

Comment:
> then a Router cluster, containing the list of all the TCP workers end up returning no address and the Context never gets closed

Just to make sure I'm following up, this change here fixes the problem above?

Reply:
Yes

Reply:
Actually, I'm fixing this method. It is incorrect as it is because it wouldn't skip to the next non-empty cluster.

Reply:
@adrianbenavides I pushed a fix and a test to make sure that `next_cluster` really does what it should.

Reply:
Nice catch there! It makes perfect sense to me now